.github/scripts/validate_pr_formatting.py

```diff
@@ -0,0 +1,344 @@
+#!/usr/bin/env python3
+#
+# Pull Request Formatting Validator
+#
+# This script validates basic pull request formatting requirements and can
+# optionally post validation comments.
+#
+# Validation Checks Performed:
+#
+# 1. PR Title Validation:
+#    - Ensures the PR title is not empty or whitespace-only
+#    - Failure comment: "⚠️ Pull request title cannot be empty."
+#
+# 2. PR Body Validation:
+#    - Ensures the PR body is not empty, null, or whitespace-only
+#    - Ensures the PR body meets minimum length requirement (147 characters)
+#      - The minimum length is based on the PR template with empty sections
+#    - Empty body failure comment: "⚠️ Add a meaningful pull request
+#      description using the PR template."
+#    - Short body failure comment: "⚠️ Provide a more detailed pull request
+#      description using the PR template (current: X characters)."
+#
+# 3. PR Template Line Validation:
+#    - Checks that template placeholder lines are removed from the PR
+#      description
+#       - Based on present template lines in the PR template file
+#    - Failure comment: "⚠️ Remove the following template lines from your
+#      PR description:" followed by list of remaining template lines
+#
+# The same comment (based on a body content hash) will not be posted multiple
+# times to avoid spamming the PR with duplicate messages.
+#
+# Copyright (c) Microsoft Corporation.
+# SPDX-License-Identifier: BSD-2-Clause-Patent
+#
+
+import base64
+import hashlib
+import os
+import re
+import sys
+from typing import List, Optional
+
+import GitHub
+
+
+class PRValidator:
+    """Logic for validating pull request formatting."""
+
+    VALIDATION_HEADER = "Pull Request Formatting Issues"
+    MIN_BODY_LENGTH = 147
+    TEMPLATE_LINE_PATTERN = r'^<_.*_>$'
+    GITHUB_PR_TEMPLATE_PATH = ".github/pull_request_template.md"
+    GITHUB_PR_TEMPLATE_URL = ("https://github.com/tianocore/edk2/blob/master/"

```

Comment:
I think you can use github API to get the repo URL path and append GITHUB_PR_TEMPLATE_PATH.  That would make this PR format checker reusable for other repos.